src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -150,21 +150,21 @@ protected TestResult parse(String expandedTestResults, AbstractBuild build, Laun
     @Override
     public void perform(Run build, FilePath workspace, Launcher launcher,
             TaskListener listener) throws InterruptedException, IOException {
-        TestResultAction action = parseAndAttach(this, null, null, build, workspace, launcher, listener);
+        TestResultAction action = parseAndAttach(this, null, null, null, build, workspace, launcher, listener);
 
         if (action != null && action.getResult().getFailCount() > 0)
             build.setResult(Result.UNSTABLE);
     }
 
     public static TestResultAction parseAndAttach(@Nonnull JUnitTask task, @CheckForNull String nodeId,
-                                                  List<String> enclosingBlocks, Run build, FilePath workspace,
-                                                  Launcher launcher, TaskListener listener)
+                                                  List<String> enclosingBlocks, List<String> enclosingBlockNames,
```

Comment:
This does not seem like a sustainable API. Introduce a new (mutable-until-“frozen”? builder?) struct type for the arguments, so that we can compatibly add information to the parser without needing to manipulate massive argument lists like this.

Reply:
*nod*